### DISCUSSION TITLE
Is there any way to query document content given the document link?

### DISCUSSION THREAD

**Discussion by kawpii** [cedricdelpoux/gatsby-source-google-docs#144] [Q&A] [answered]
As mentioned in the title I need to fetch the content of a document given the document URL. Is there anyway to achieve this?

**Comment by cedricdelpoux** [comment]
You can put your document in a folder and query this specific folder

**Reply by kawpii** [reply]
No what I want is to query the document by URL, which is in a google sheet's cell

**Reply by cedricdelpoux** [reply]
Unfortunately, you can't do that with `gatsby-source-google-docs`. PR are welcome

**Reply by kawpii** [reply]
How do you suggest implementing this? To me it seems like your plugin is designed in a way to support page creation from set of docs.

**Reply by kawpii** [reply]
But there may be way since you must be already using document ids, do you have any ideas?

**Comment by cedricdelpoux** [comment]
I suggest to modify the function `fetchFiles` (https://github.com/cedricdelpoux/gatsby-source-google-docs/blob/4995e89160abdd4e11ea126e7fdeb4d76f557222/utils/google-drive.js#L306) 

If there is a "urls" option, then fetch by url, otherwise do the default behaviour with the `folder` option.

**Reply by kawpii** [reply]
Lets say we have new option called `url:'true'` then what? How can I get transformed document content after getting the URLs from a Google Sheet cells

**Reply by cedricdelpoux** [reply]
I think your usecase is too specific. Could you tell me why you put documents urls in a Google sheet cell ?

**Reply by kawpii** [reply]
Since Google Sheets do not support rich text and it is not designed for rich text, so I want use Docs to support rich text. I am using https://github.com/butlerx/gatsby-source-google-spreadsheets to fetch data from Google Sheets.

**Reply by cedricdelpoux** [reply]
But why not using Google docs directly? Google Sheets is not designed to store Google Docs urls

**Reply by kawpii** [reply]
Each row in Google Sheet have information associated with an item, but since Google Sheets do not support rich text fully, I need a way to store rich text associated with a particular item, that is why I need to store those info in Google Docs.

**Reply by cedricdelpoux** [reply]
With gatsby-source-google-docs, you can attach extra data to each documents using document description field using YAML format

**Reply by kawpii** [reply]
So you are suggesting to add query values from the Sheet and set those data to description field in the document?

**Reply by cedricdelpoux** [reply]
I suggest you to use metadata if you need to attach more data to your documents. 
Checkout https://cedricdelpoux.github.io/gatsby-source-google-docs/metadata for examples.
In this case, you don't need Google Sheet

**Reply by kawpii** [reply]
But the problem is I want to allow users (content creators) to update data to through Sheets since they are already familiar with that.
It seems there is not much you can with my usecase. 

Can you please tell me how you parse and transform Docs content to html markup? When I tried the Docs REST API there was lot of JSON attributes with lots of stuff in the response which is tedius to parse. Can you please share how you do that?

**Reply by kawpii** [reply]
@cedricdelpoux Can you please tell me how you parse and transform Docs content to html markup? When I tried the Docs REST API there was lot of JSON attributes with lots of stuff in the response which is tedius to parse. Can you please share how you do that?

**Comment by cedricdelpoux** [comment, marked as answer]
You can do 
```
import { GoogleDocument } from 'gatsby-source-google-docs/utils/google-document'

const doc = new GoogleDocument(docsResponse);
const markdown = doc.toMarkdown()
```

**Reply by kawpii** [reply]
Awesome! Thanks a lot! 

Then I have to just use `gatsby-transformer-remark` plugin to convert markdown to HTML!

**Reply by cedricdelpoux** [reply]
That's right.
Please share your work when you are done. It could gives ideas to some people

**Reply by cedricdelpoux** [reply]
@kaushalyap Can I checkout you code somewhere to see how you used gatsby-source-google-docs? It's interesting

**Reply by kawpii** [reply]
@cedricdelpoux sorry to let you know, actually I did not use it and left Gatsby.

**Reply by cedricdelpoux** [reply]
ok thanks!
